x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/EngineIndexService.java

```diff
@@ -276,23 +305,18 @@ private void updateEngine(Engine engine, ActionListener<IndexResponse> listener)
         }
     }
 
-    /**
-     * Deletes the provided {@param engineName} in the underlying index, or delegate a failure to the provided
-     * listener if the resource does not exist or failed to delete.
-     *
-     * @param engineName The name of the {@link Engine} to delete.
-     * @param listener The action listener to invoke on response/failure.
-     *
-     */
-    public void deleteEngine(String engineName, ActionListener<DeleteResponse> listener) {
-        try {
-            // TODO Delete alias when Engine is deleted
-            final DeleteRequest deleteRequest = new DeleteRequest(ENGINE_ALIAS_NAME).id(engineName)
-                .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
-            clientWithOrigin.delete(deleteRequest, listener);
-        } catch (Exception e) {
-            listener.onFailure(e);
-        }
+    private static SearchEnginesResult mapSearchResponse(SearchResponse response) {
+        Engine[] engines = Arrays.stream(response.getHits().getHits()).map(EngineIndexService::hitToEngine).toArray(Engine[]::new);
+        return new SearchEnginesResult(engines, (int) response.getHits().getTotalHits().value);
+    }
+
+    private static Engine hitToEngine(SearchHit searchHit) {
```

Comment:
I am not sure we should use the `Engine` class in the response of this API. We won't return the entire `Engine` object, it's possible currently because we only have a name, a list of indices and an analytics collection name but in the near future we'll have more configurations and they won't be returned in the list API.

Reply:
Right now the response returns the entire Engine. If the EngineIndexService does not retrieve the Engines, then the Transport action will need to do so, right?

Reply:
I don't think we should return the entire Engine here. The list API can be used to list the engine names, their associated indices and maybe their analytics collections names. To get the entire Engine configuration, another call to get the Engine by name should be needed (using the get API).

Reply:
Can we create a special class for this like `EngineListResponse`? As I said in my previous comment I'd like to avoid returning the entire Engine. It might be ok now but we'll add more configurations later and we shouldn't return the full object in this API. Let's restrict the response to the indices names, the alias name and the analytics collection name? I know that it's the entire Engine configuration now but this will change in the near future.

Reply:
Done! LMKWYT